cmd/backup-manager/app/clean/manager.go

```diff
@@ -135,22 +135,35 @@ func (bm *Manager) getNextBackup(ctx context.Context, backup *v1alpha1.Backup) (
 
 	for i, bk := range bks {
 		if backup.Name == bk.Name {
-			// only one backup or it is the last one in slice
-			if i >= len(bks)-1 {
-				return nil, nil
-			}
-			return bks[i+1], nil
+			return bm.getFirstVolumeSnapshotBackup(bks[i:])
```

Comment:
Why not use `bks[i+1:]`?

Reply:
I+1 may overflow cause panic, and the backup list bks contain non-volume snapshot backup

Reply:
fixed in https://github.com/pingcap/tidb-operator/pull/4819/commits/e963c3dce6ed5891897a7b1e1a6098e4209c82d5